File: golang/invert_binary_tree.go
package golang

import (
	"golang/model"
)

func invertTree(root *model.TreeNode) *model.TreeNode {
	p := root

	if p != nil {
		p.Left, p.Right = p.Right, p.Left
		invertTree(p.Left)
		invertTree(p.Right)
	}

	return root
}

File: python3/invert_binary_tree.py
from .model import TreeNode

# Time  : O(n)
# Space : O(1)

class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        p = root
        
        if p:
            p.left, p.right = p.right, p.left
            self.invertTree(root.left)
            self.invertTree(root.right)
            
        return root
